### ISSUE TITLE
Primary constructor is linted by prefer-static-class

### ISSUE THREAD

**Issue by ngoc-quoc-huynh** [bancolombia/dart-code-linter#272]
The new introduced primary constructor in Dart 3.13.0 counts it as a static/global function and therefor lints it as `prefer-static-class` .

```dart
class Foo(final int n) {  // Lints: Prefer declaring static class members instead of global functions, variables and constants
  void bar() => print(n);
}
```

This rule should not be triggered by the primary constructor.

**Comment by juan-campuzano** [comment]
@ngoc-quoc-huynh nice catch, would you like to contribute with a fix maybe? currently working on a new metric

**Comment by ngoc-quoc-huynh** [comment]
@juan-campuzano I can do that, but I'll be on vacation for one month. Maybe I can find some time in the next few days.
